### PR TITLE
Improve Jenkins CI

### PR DESCRIPTION
**Pull request by tul1**
I've added a brand new pipeline that is going to use our Docker image and perform more jobs in order to validate the PR's code before enabling the merge operation into master. Also a publishing stage was added, allowing to deploy avalon-rs lib into crate.io when merging from master into stable.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,29 +1,61 @@
...
+        stage('Build') {
+            steps {
+                sh "cargo build"
+            }
+        }
+        stage('Test') {
+            steps {
+                // TODO uncomment once cargo test passes. 
+                // sh "cargo test"
+                sh "echo coming soon"
```
Comment (julianmateu) [inline Jenkinsfile]: Are we merging this if the test are not running? Can't we make this a draft PR until that is solved so we don't loose our existing CI?

File: src/core/election.rs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+pub struct Scrutiny<T> {
+    result: HashMap::<Option<T>, usize>,
+}
+
+impl<T> Scrutiny<T> {
+    pub fn result(&self) -> &HashMap::<Option<T>, usize> {
+        &self.result
+    }
+}
```
Comment (julianmateu) [inline src/core/election.rs]: These files shouldn't be in the PR, maybe rebase?

File: Jenkinsfile (hunk 1 above)
Comment (tul1) [reply Jenkinsfile]: Tests were solved inside this pr https://github.com/tul1/avalon-rs/pull/28 . we can wait until it's merge. You cannot make a pr a draft pr.

File: src/core/election.rs (hunk 2 above)
Comment (tul1) [reply src/core/election.rs]: indeed, i'll rebase

File: Jenkinsfile (hunk 1 above)
Comment (julianmateu) [reply Jenkinsfile]: OK. Let's wait until that one is merged

Comment (tul1) [reply Jenkinsfile]: uncommneted

**Review by julianmateu — APPROVED** [review]

**Pull request merged**